impeller/entity/contents/runtime_effect_contents.cc

```diff
@@ -206,84 +204,111 @@ bool RuntimeEffectContents::Render(const ContentContext& renderer,
   /// Fragment stage uniforms.
   ///
 
-  size_t minimum_sampler_index = 100000000;
-  size_t buffer_index = 0;
-  size_t buffer_offset = 0;
-  for (const auto& uniform : runtime_stage_->GetUniforms()) {
-    std::shared_ptr<ShaderMetadata> metadata = MakeShaderMetadata(uniform);
-
-    switch (uniform.type) {
-      case kSampledImage: {
-        // Sampler uniforms are ordered in the IPLR according to their
-        // declaration and the uniform location reflects the correct offset to
-        // be mapped to - except that it may include all proceeding float
-        // uniforms. For example, a float sampler that comes after 4 float
-        // uniforms may have a location of 4. To convert to the actual offset we
-        // need to find the largest location assigned to a float uniform and
-        // then subtract this from all uniform locations. This is more or less
-        // the same operation we previously performed in the shader compiler.
-        minimum_sampler_index =
-            std::min(minimum_sampler_index, uniform.location);
-        break;
-      }
-      case kFloat: {
-        size_t alignment =
-            std::max(uniform.bit_width / 8, DefaultUniformAlignment());
-        auto buffer_view = pass.GetTransientsBuffer().Emplace(
-            uniform_data_->data() + buffer_offset, uniform.GetSize(),
-            alignment);
-
-        ShaderUniformSlot uniform_slot;
-        uniform_slot.name = uniform.name.c_str();
-        uniform_slot.ext_res_0 = uniform.location;
-        cmd.BindResource(ShaderStage::kFragment, uniform_slot, metadata,
-                         buffer_view);
-        buffer_index++;
-        buffer_offset += uniform.GetSize();
-        break;
-      }
-      case kBoolean:
-      case kSignedByte:
-      case kUnsignedByte:
-      case kSignedShort:
-      case kUnsignedShort:
-      case kSignedInt:
-      case kUnsignedInt:
-      case kSignedInt64:
-      case kUnsignedInt64:
-      case kHalfFloat:
-      case kDouble:
-        VALIDATION_LOG << "Unsupported uniform type for " << uniform.name
-                       << ".";
-        return true;
-    }
-  }
+  // size_t minimum_sampler_index = 100000000;
+  // size_t buffer_index = 0;
+  // size_t buffer_offset = 0;
+
+  ShaderUniformSlot uniform_slot;
+  uniform_slot.name = "_RESERVED_IDENTIFIER_FIXUP_gl_DefaultUniformBlock";
+  uniform_slot.ext_res_0 = 0;
+  uniform_slot.binding = 64;
+  ShaderMetadata metadata;
+  metadata.name = uniform_slot.name;
+  auto uniform_count = runtime_stage_->GetUniforms().size();
+  metadata.members.emplace_back(ShaderStructMemberMetadata{
+      .type = ShaderType::kStruct,
+      .size = uniform_count * 4,
+      .byte_length = uniform_count * 4 / 8,
+  });
+
```

Comment:
Just to narrow things down, the metadata is only used by GLES. Vulkan only cares about binding and descriptor set layout